subsys/net/lib/mqtt_helper/Kconfig

```
@@ -85,7 +85,7 @@ config MQTT_HELPER_PAYLOAD_BUFFER_LEN
 
 config MQTT_HELPER_PROVISION_CERTIFICATES
 	bool "Run-time provisioning of certificates"
-	depends on (BOARD_QEMU_X86 || BOARD_NATIVE_POSIX || BOARD_NRF7002DK_NRF5340_CPUAPP || BOARD_NRF7002DK_NRF5340_CPUAPP_NS) && MQTT_LIB_TLS
+	depends on (BOARD_QEMU_X86 || BOARD_NATIVE_SIM || BOARD_NRF7002DK_NRF5340_CPUAPP || BOARD_NRF7002DK_NRF5340_CPUAPP_NS) && MQTT_LIB_TLS
```

I think updating our tests to use `native_sim` is a good improvement. Though I wonder if we should continue to support `native_posix` as a board in NCS, at least for now. Customers might have tests running with `native_posix` board for their applications. Do we want to force them to change to `native_sim` as well?

```suggestion
	depends on (BOARD_QEMU_X86 || BOARD_NATIVE_SIM || BOARD_NATIVE_POSIX || BOARD_NRF7002DK_NRF5340_CPUAPP || BOARD_NRF7002DK_NRF5340_CPUAPP_NS) && MQTT_LIB_TLS
```

---

@aescolar could you weigh in please?

---

Apart from https://github.com/nrfconnect/sdk-nrf/pull/16584#discussion_r1688122345 ;)
It is up to somebody else to decide.
Anyhow, you can expect that in ~2 weeks I will deprecate native_posix in upstream Zephyr. So by the next "upmerge" the deprecation will be propagated to ncs.